src/HealthCheck/DoctrineConnectionHealthCheck.php

```diff
@@ -30,34 +31,28 @@
  */
 class DoctrineConnectionHealthCheck implements HealthCheckInterface
 {
-    /**
-     * @var EntityManager|null
-     */
-    private $entityManager;
 
-    /**
-     * @param EntityManager $entityManager
-     */
-    public function setEntityManager(EntityManager $entityManager)
-    {
-        $this->entityManager = $entityManager;
+    public function __construct(
+        #[Autowire(service: 'doctrine.dbal.default_connection')]
+        private readonly ?Connection $connection
+    ) {
     }
 
     public function check(HealthReportInterface $report): HealthReportInterface
     {
         // Was the entityManager injected? When it is not the project does not use Doctrine ORM
```

Comment:
1. Please update the doc block here. This might cause confusion in the future as ORM is no longer used. 
3. Also, maybe test in the constructor if the service was set correctly? That way the app halts at an earlier stage, and your control logic is cleaner

Reply:
Hmmm, what should be the behaviour then if the connection is not there at constructor time? We cannot return $report then, like it is now...

Reply:
Not sure how to move the connection check to the constructor and preserve behaviour? Any suggestions?

Reply:
You are right we explicitly allow for the service to be null. No use to check it in the constructor then..

Leave it like it is right now!